False positive on Rule "no-import-all-from-library"

### Current Behavior

When I import **few functions** from lodash, the linter raise an issue telling

> **You not import all from library lodash**

<img width="1898" height="314" alt="Image" src="https://github.com/user-attachments/assets/111c901c-ac97-4f71-bb3b-963c7657584c" />

### Expected Behavior

The code is respecting the rule, so it should not raise an issue here

### Steps to Reproduce

1. activate `@credengo/eslint-plugin` version 2.1
2. add the code `import { memoize, omitBy, isNil } from 'lodash'`
3. let the linter analyse the code
4. See the raised issues

### Failure Logs

```shell
> eslint src --fix


/......../src/api/my-code.js
  18:1  warning  You should not import all from library lodash  @creedengo/no-import-all-from-library
```

### creedengo-javascript Version

2.1

### SonarQube Version

_No response_

### Additional Information

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- [#112](https://github.com/green-code-initiative/creedengo-javascript/pull/112) Clarify rule GCI9 "no-import-all-from-library"
 - [#113](https://github.com/green-code-initiative/creedengo-javascript/pull/113) Extend rule GCI530 "no-torch" to detect HTML5 Web API usage
 
 ## [3.1.0] - 2026-05-10

--- a/eslint-plugin/lib/rules/no-import-all-from-library.js
+++ b/eslint-plugin/lib/rules/no-import-all-from-library.js
@@ -84,7 +84,13 @@ module.exports = {
       ImportDeclaration(node) {
         const currentLibrary = node.source.value;
 
-        const forbiddenByName = notAllowedLibraries.includes(currentLibrary);
+        const forbiddenByName =
+          notAllowedLibraries.includes(currentLibrary) &&
+          node.specifiers.some(
+            (specifier) =>
+              specifier.type === "ImportDefaultSpecifier" ||
+              specifier.type === "ImportNamespaceSpecifier",
+          );
         const forbiddenByNamespace =
           importByNamespaceNotAllowedLibraries.includes(currentLibrary) &&
           node.specifiers.some(

--- a/eslint-plugin/lib/rules/no-import-all-from-library.js
+++ b/eslint-plugin/lib/rules/no-import-all-from-library.js
@@ -28,8 +28,10 @@ module.exports = {
       recommended: "warn",
     },
     messages: {
-      ShouldNotImportAllFromLibrary:
-        "You should not import all from library {{library}}",
+      doNotImportFromLibrary:
+        "Avoid importing from the main library path {{library}}, use specific subpaths instead",
+      doNotUseNamespaceImport:
+        "Avoid namespace imports from {{library}}, use named imports instead",
     },
     schema: [
       {
@@ -74,9 +76,9 @@ module.exports = {
       }
 
       if (option.importByNamespaceNotAllowedLibraries) {
-        notAllowedLibraries.push(
-          ...option.importByNamespaceNotAllowedLibraries,
-        );
+        for (const lib of option.importByNamespaceNotAllowedLibraries) {
+          importByNamespaceNotAllowedLibraries.add(lib);
+        }
       }
     }
 
@@ -84,23 +86,25 @@ module.exports = {
       ImportDeclaration(node) {
         const currentLibrary = node.source.value;
 
-        const forbiddenByName =
-          notAllowedLibraries.includes(currentLibrary) &&
-          node.specifiers.some(
-            (specifier) =>
-              specifier.type === "ImportDefaultSpecifier" ||
-              specifier.type === "ImportNamespaceSpecifier",
-          );
+        const forbiddenByName = notAllowedLibraries.includes(currentLibrary);
         const forbiddenByNamespace =
           importByNamespaceNotAllowedLibraries.has(currentLibrary) &&
           node.specifiers.some(
             (specifier) => specifier.type === "ImportNamespaceSpecifier",
           );
 
-        if (forbiddenByName || forbiddenByNamespace) {
+        if (forbiddenByName) {
           context.report({
             node,
-            messageId: "ShouldNotImportAllFromLibrary",
+            messageId: "doNotImportFromLibrary",
+            data: { library: currentLibrary },
+          });
+        }
+
+        if (forbiddenByNamespace) {
+          context.report({
+            node,
+            messageId: "doNotUseNamespaceImport",
             data: { library: currentLibrary },
           });
         }

--- a/eslint-plugin/tests/lib/rules/no-import-all-from-library.test.js
+++ b/eslint-plugin/tests/lib/rules/no-import-all-from-library.test.js
@@ -54,6 +54,12 @@ const tests = {
     `
     import map from 'underscore/modules/map.js';
     `,
+    `
+    import { memoize, omitBy, isNil } from 'lodash';
+    `,
+    `
+    import { isEmpty } from 'underscore';
+    `,
   ],
 
   invalid: [

--- a/eslint-plugin/tests/lib/rules/no-import-all-from-library.test.js
+++ b/eslint-plugin/tests/lib/rules/no-import-all-from-library.test.js
@@ -36,9 +36,9 @@ const ruleTester = new RuleTester({
     sourceType: "module",
   },
 });
-const expectedError = {
-  messageId: "ShouldNotImportAllFromLibrary",
-};
+
+const doNotImportFromLibraryError = { messageId: "doNotImportFromLibrary" };
+const doNotUseNamespaceImportError = { messageId: "doNotUseNamespaceImport" };
 
 const tests = {
   valid: [
@@ -54,36 +54,30 @@ const tests = {
     `
     import map from 'underscore/modules/map.js';
     `,
-    `
-    import { memoize, omitBy, isNil } from 'lodash';
-    `,
-    `
-    import { isEmpty } from 'underscore';
-    `,
   ],
 
   invalid: [
     {
       code: "import lodash from 'lodash';",
-      errors: [expectedError],
+      errors: [doNotImportFromLibraryError],
     },
     {
       code: "import * as lodash from 'lodash';",
-      errors: [expectedError],
+      errors: [doNotImportFromLibraryError],
     },
     {
       code: "import * as lodash from 'lodash-es';",
-      errors: [expectedError],
+      errors: [doNotUseNamespaceImportError],
     },
     {
       code: "import someLib from 'some-lib';",
       options: [{ notAllowedLibraries: ["some-lib"] }],
-      errors: [expectedError],
+      errors: [doNotImportFromLibraryError],
     },
     {
       code: "import * as someLib from 'some-lib';",
       options: [{ importByNamespaceNotAllowedLibraries: ["some-lib"] }],
-      errors: [expectedError],
+      errors: [doNotUseNamespaceImportError],
     },
   ],
 };

--- a/test-project/src/modular-import-from-library.js
+++ b/test-project/src/modular-import-from-library.js
@@ -1,3 +1,9 @@
+import { orderBy } from "lodash"; // Non-compliant: import of the main path of a bad library
+
+import * as _ from "underscore"; // Non-compliant: import by namespace of a bad library
+
 import isEmpty from "lodash/isEmpty"; // Compliant
 
+orderBy([], [], []);
+_.allKeys([]);
 isEmpty("");

